streets_reader/src/osm_reader/clip.rs

```diff
@@ -0,0 +1,89 @@
+use geom::{Distance, PolyLine, Polygon};
+
+use super::Document;
+
+impl Document {
+    // TODO This destroys the guarantee that the Document represents raw OSM. Do we need to be
```

Comment:
Depending what else we do with `Document`, we have to be careful here. `way.pts` and `way.nodes` get out of sync. Nothing downstream uses `way.nodes` right now, but if that ever changes...